File: Python_lab1/0517054_Q1.py

# coding: utf-8

# In[7]:


import random
import math


# In[10]:


N = int(input())
points_inside = 0
for i in range(N):
    x = random.random()
    y = random.random()
    d = math.sqrt(x**2 + y**2)
    if d <= 1:
        points_inside += 1
pi = 4 * (points_inside / N)
print(pi)

File: lab5/0517054_Q1.c
#include <stdio.h>
int main(){
    int input, a, times = 1;
    while (1){
        printf("Input: ");
        scanf("%d", &input);
    
        if (2 <= input && input <= 10000){
            printf("%d = ",input);
            for (int i = 2; i <= input; i++){
            
                if (input % i == 0){
                    printf("%d", i);
                    input /= i;

                    while ( input % i == 0 ){
                        times += 1;
                        input /= i;
                    }
                    if (times > 1)
                        printf("^%d", times);
                    times = 1;
               
                    if (input > 1)
                        printf(" * ");
                    else
                        break;
                }
            }
            printf("\n");
        }
        else if (input == 0)
            break;
        else
            printf("Out of range.\n");
        
    }
    return 0;
}
